main/MiNiFiMain.cpp

```diff
@@ -208,6 +238,10 @@ int main(int argc, char **argv) {
   configure->setHome(minifiHome);
   configure->loadConfigureFile(DEFAULT_NIFI_PROPERTIES_FILE);
 
+  if (containsEncryptedProperties(*configure)) {
+    decryptSensitiveProperties(*configure, minifiHome, *logger);
```

Comment:
as I see, we update the `configure` object in `decryptSensitiveProperties `, one possible problem with this, is that on a flow update (from a C2 agent) the configurations file is persisted by default (to store the new flow url) this could inadvertently leak the decrypted values

Reply:
Ouch.  Yes, that is a serious problem.

Reply:
fixed in https://github.com/apache/nifi-minifi-cpp/pull/914/commits/10259562e26b6be5c8dbd6ba200118dc43f0d1e5